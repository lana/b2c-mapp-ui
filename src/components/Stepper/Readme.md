# Stepper

A step based progress indicator for multi screen flows.

| Prop         | Value Type | Required |
| ------------ | ---------- | -------- |
| `className`  | `String`   | false    |
| `dataTestId` | `String`   | false    |
| `title`      | `String`   | false    |
| `steps`      | `Int`      | **true** |
| `activeStep` | `Int`      | false    |
| `hideActive` | `Bool`     | false    |

```
<Stepper title='Your progress' steps={4} activeStep={1} />
```
