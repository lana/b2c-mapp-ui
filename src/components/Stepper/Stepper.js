import Heading from '../Heading/Heading.vue';

const components = {
  Heading,
};

const props = {
  dataTestId: {
    type: String,
    default: 'navigation-stepper',
  },
  title: {
    type: String,
    default: '',
  },
  hideActiveStep: Boolean,
  modelValue: Number,
  countOfSteps: Number,
};

const computed = {
  availableSteps() {
    if (!this.countOfSteps) { return []; }
    const result = [...Array(this.countOfSteps).keys()].map((step, index) => ({
      isActive: ((index === this.modelValue) && !this.hideActiveStep),
      isInactive: (index > this.modelValue), // WARNING: `Inactive` in this context is not the same thing as `!isActive`
      stepNumber: (index + 1),
    }));
    return result;
  },
};

const name = 'Stepper';

const Stepper = {
  name,
  components,
  props,
  computed,
};

export default Stepper;
