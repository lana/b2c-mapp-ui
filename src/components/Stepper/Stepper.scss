@import '../../styles/colors';

.stepper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  height: 8px;
  padding: 2px 16px;
  text-align: center;
}

.steps {
  display: inline-block;
  height: 20px;
}

.step {
  display: inline-block;
  vertical-align: middle;
  width: 16px;
  height: 4px;
  margin: 0 4px;
  border-radius: 2px;
  background-color: $blue-500;
  font: 0/0 a;
  transition: all 0.25s ease-out;
}

.active {
  background-color: $blue-500; // TODO: Should we change this color so that it stands out from the default step color? (if not: then we should remove the hideActiveStep prop since it currently does nothing)
}

.inactive {
  background-color: $blue-700;
  opacity: 0.2;
}
