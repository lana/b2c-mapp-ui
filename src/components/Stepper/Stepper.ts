import { defineComponent } from 'vue';

import Heading from '../Heading/Heading.vue';

const Stepper = defineComponent({
  name: 'Stepper',
  components: {
    Heading,
  },
  props: {
    dataTestId: {
      type: String,
      default: 'navigation-stepper',
    },
    title: {
      type: String,
      default: '',
    },
    hideActiveStep: Boolean,
    modelValue: {
      type: Number,
      default: 0,
    },
    countOfSteps: {
      type: Number,
      default: 0,
    },
  },
  computed: {
    availableSteps() {
      if (!this.countOfSteps) { return []; }
      const result = Array(this.countOfSteps).fill(0).map((_step, index) => ({
        isActive: ((index === this.modelValue) && !this.hideActiveStep),
        isInactive: (index > this.modelValue), // WARNING: `Inactive` in this context is not the same thing as `!isActive`
        stepNumber: (index + 1),
      }));
      return result;
    },
  },
});

export default Stepper;
