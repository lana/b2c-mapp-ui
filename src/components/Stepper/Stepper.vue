<template>
  <nav :data-testid="`${dataTestId}-navigation`" class="stepper">
    <Heading v-if="title" type="callout">
      {{ title }}
    </Heading>
    <ol :data-testid="`${dataTestId}-list`" class="steps">
      <li v-for="({ stepNumber, isActive, isInactive }) in availableSteps"
          :key="stepNumber"
          class="step"
          :class="{ active: isActive, inactive: isInactive }"
          :data-testid="`${dataTestId}-element`"
      >
        Paso {{ stepNumber }}
      </li>
    </ol>
  </nav>
</template>

<script lang="ts" src="./Stepper.ts"></script>
<style lang="scss" scoped src="./Stepper.scss"></style>
