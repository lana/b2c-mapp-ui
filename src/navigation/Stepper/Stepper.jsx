import CSS from './styles.css';
import Heading from '../../typography/Heading/Heading';

export default ({ steps, activeStep, hideActive, title, className, dataTestId }) => {
	const testId = dataTestId || 'navigation-stepper';

	return (
		<navigation data-testid={`${testId}-navigation`} className={`${CSS.stepper} ${className}`}>
			{title && <Heading type={`callout`}>{title}</Heading>}
			<ol data-testid={`${testId}-list`} className={CSS.steps}>
				{[...Array(steps).keys()].map((step, idx) => {
					let activeClass = idx === activeStep && !hideActive ? CSS.active : '';
					let inactiveClass = idx > activeStep ? CSS.inactive : '';
					let index = idx + 1;
					return (
						<li data-testid={`${testId}-element`} className={`${CSS.step} ${activeClass} ${inactiveClass}`}>
							Paso {index}
						</li>
					);
				})}
			</ol>
		</navigation>
	);
};
