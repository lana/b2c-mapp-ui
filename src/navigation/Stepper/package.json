{
  "name": "Stepper",
  "version": "0.0.0",
  "private": true,
  "main": "./Stepper.jsx"
}
