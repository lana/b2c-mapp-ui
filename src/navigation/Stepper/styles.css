.stepper {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	flex-shrink: 0;
	height: 56px;
	padding: 0 16px;
	text-align: center;
}

.steps { display: inline-block; height: 20px; }

.step {
	display: inline-block;
	vertical-align: middle;
	width: 24px;
	height: 4px;
	margin: 0 4px;
	border-radius: 2px;
	background-color: var(--blue);
	font: 0/0 a;
	transition: all 0.25s ease-out;
}

.active { background-color: var(--blue); }

.inactive {
	background-color: var(--dblue);
	opacity: 0.2; 
}
