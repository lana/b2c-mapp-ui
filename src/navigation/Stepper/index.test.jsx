import { mount } from 'enzyme';
import Stepper from './index'
import { shallow } from 'preact-render-spy';
import Router from '../../../../src/Router'

describe('UI/navigation/Stepper', () => {
	let wrapper;
	test('should render correctly',  () => {
		wrapper = mount(<Stepper steps={3} activeStep={2} hideActive={false}  className="teste" title="teste" />)
		expect(wrapper).toMatchSnapshot()
	})
	test('should render correctly',  () => {
		wrapper = mount(<Stepper steps={3} activeStep={1} />)
		expect(wrapper).toMatchSnapshot()
	})
	test('should find three steps',  () => {
		const div = document.createElement('div')
		let arr;
		wrapper = mount(<Stepper steps={3} activeStep={1} />)
		div.innerHTML = wrapper.html()
		arr = [].slice.call(div.querySelectorAll('li'))
		expect(arr.length).toBe(3)
		expect(div.querySelector('li:nth-child(2)').className).toContain('active')

	})
})
