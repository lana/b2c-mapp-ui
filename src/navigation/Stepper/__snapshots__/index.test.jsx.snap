// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/navigation/Stepper should render correctly 1`] = `ReactWrapper {}`;

exports[`UI/navigation/Stepper should render correctly 2`] = `ReactWrapper {}`;
